### PR TITLE
F8 related changes for naive convolution solver and tests [1/5]

### PR DESCRIPTION
**Pull request by xinlipn**
This PR contains new rocBLAS API changes that support F8 type

### CODE REVIEW

File: src/include/miopen/solver.hpp [hunk 1]
```diff
@@ -415,6 +415,42 @@ struct ConvAsm1x1U final : ConvTunableSolver<PerformanceConfigConvAsm1x1U>
...
+    bool IsValidPerformanceConfig(const ConvolutionContext&,
+                                  const ProblemDescription&,
+                                  const PerformanceConfigConvBiasActivAsm1x1U&) const override;
+    PerformanceConfigConvBiasActivAsm1x1U Search(const ConvolutionContext&,
+                                                 const ProblemDescription&,
+                                                 const AnyInvokeParams& invoke_ctx) const override;
+    ConvSolution GetSolution(const ConvolutionContext& params,
+                             const ProblemDescription& problem,
+                             const PerformanceConfigConvBiasActivAsm1x1U& config) const override;
+};
```
Comment (JehandadKhan) [inline src/include/miopen/solver.hpp]: This change should not be here

Comment (JehandadKhan) [reply src/include/miopen/solver.hpp]: Perhaps a merge conflict, maybe merging the base branch into this branch will fix this issue.

File: src/include/miopen/solver.hpp [hunk 2]
```diff
@@ -1167,6 +1203,50 @@ struct ConvMlirIgemmFwdXdlops final : ConvTunableSolver<PerformanceConvMlirIgemm
                              const PerformanceConvMlirIgemmXdlops&) const override;
 };
 
+struct PerformanceImplicitGemmV4R4GenXdlopsFwdFp32
```
Comment (JehandadKhan) [inline src/include/miopen/solver.hpp]: Same as above

File: src/include/miopen/solver.hpp [hunk 3]
```diff
@@ -1236,6 +1316,57 @@ struct ConvMlirIgemmWrWXdlops final : ConvTunableSolver<PerformanceConvMlirIgemm
                              const PerformanceConvMlirIgemmXdlops&) const override;
 };
 
+struct PerformanceImplicitGemmXdlops : PerfConfigBase<PerformanceImplicitGemmXdlops>
```
Comment (JehandadKhan) [inline src/include/miopen/solver.hpp]: same as above

File: src/include/miopen/solver.hpp
Comment (JehandadKhan) [inline src/include/miopen/solver.hpp:1]: Please check the changes in this file, most of these changes should not be here

File: src/kernels/gpu_reference_kernel/fp8_kern_types.h [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+#ifndef ACCUMULATOR_TYPE
+#define ACCUMULATOR_TYPE double
+#endif
+
+#define KERNEL_NAME_SUFFIX CAT(CAT(INPUT_TYPE, _), CAT(CAT(WEIGHTS_TYPE, _), OUTPUT_TYPE))
+
+#define FWD_KERNEL_NAME CAT(naive_conv_fwd_nchw_, KERNEL_NAME_SUFFIX)
+#define BWD_KERNEL_NAME CAT(naive_conv_bwd_nchw_, KERNEL_NAME_SUFFIX)
+#define WRW_KERNEL_NAME CAT(naive_conv_wrw_nchw_, KERNEL_NAME_SUFFIX)
```
Comment (JehandadKhan) [inline src/kernels/gpu_reference_kernel/fp8_kern_types.h]: Please add a new line at the end of the file

File: src/kernels/gpu_reference_kernel/fp8_kern_types.h [hunk 5]
```diff
@@ -0,0 +1,38 @@
+#pragma once
```
Comment (JehandadKhan) [inline src/kernels/gpu_reference_kernel/fp8_kern_types.h:26]: Please check all new files for copyright notice.

File: src/kernels/gpu_reference_kernel/fp8_naive_conv.cpp [hunk 6]
```diff
@@ -0,0 +1,553 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2020-2021 Advanced Micro Devices, Inc.
```
Comment (JehandadKhan) [inline src/kernels/gpu_reference_kernel/fp8_naive_conv.cpp]: Please update the date

File: src/kernels/hip_f8_impl.h [hunk 7]
```diff
@@ -0,0 +1,293 @@
+//
```
Comment (JehandadKhan) [inline src/kernels/hip_f8_impl.h]: Missing copyright header

File: src/kernels/hip_float8.h [hunk 8]
```diff
@@ -0,0 +1,663 @@
...
+                             hip_bfloat16 v3         = hip_bfloat16(0.0f),
+                             miopen_f8::hip_f8_rounding_mode rm = miopen_f8::hip_f8_rounding_mode::standard,
+                             uint32_t rng            = 0);
+    HIP_HOST_DEVICE hip_f8x4(hip_bfloat16x2 v,
+                             miopen_f8::hip_f8_rounding_mode rm = miopen_f8::hip_f8_rounding_mode::standard,
+                             uint32_t rng            = 0);
+    HIP_HOST_DEVICE hip_f8x4(hip_bfloat16x4 v,
+                             miopen_f8::hip_f8_rounding_mode rm = miopen_f8::hip_f8_rounding_mode::standard,
+                             uint32_t rng            = 0);
+#endif
```
Comment (JehandadKhan) [inline src/kernels/hip_float8.h]: Please remove dead code

File: src/solver/conv_direct_naive_conv.cpp [hunk 9]
```diff
@@ -100,6 +102,23 @@ bool IsOutputInt32(const ProblemDescription& problem)
...
+    if(data_type == miopenHalf)
+        return "half";
+    if(data_type == miopenFloat8)
+        return "float8";
+    if(data_type == miopenBFloat8)
+        return "bfloat8";
+    if(data_type == miopenDouble)
+        return "double";
+    MIOPEN_THROW("Unimplemented type in FP8 kernels");
+}
```
Comment (JehandadKhan) [inline src/solver/conv_direct_naive_conv.cpp]: This function exists elsewhere in MIOpen please avoid duplicate code.

File: src/solver/conv_winoRxS.cpp [hunk 10]
```diff
@@ -381,6 +381,13 @@ bool PerformanceConfigConvBinWinogradRxS::operator==(
     return n_groups == other.n_groups;
 }
 
+std::string PerformanceConfigConvBinWinogradRxS::ToString() const
```
Comment (JehandadKhan) [inline src/solver/conv_winoRxS.cpp]: This change should not show up in the diff.

File: src/solver/conv_winoRxS.cpp [hunk 11]
```diff
@@ -555,12 +562,14 @@ static bool IsApplicableBase(const ConvolutionContext& ctx, const ProblemDescrip
         return false;
     if(problem.IsFp16() &&
        !(StartsWith(name, "gfx906") || StartsWith(name, "gfx908") || StartsWith(name, "gfx90a") ||
-         StartsWith(name, "gfx94") || StartsWith(name, "gfx1011") || StartsWith(name, "gfx1012") ||
-         StartsWith(name, "gfx103") || StartsWith(name, "gfx11")))
+         StartsWith(name, "gfx1011") || StartsWith(name, "gfx1012") || StartsWith(name, "gfx103") ||
+         StartsWith(name, "gfx11")))
```
Comment (JehandadKhan) [inline src/solver/conv_winoRxS.cpp]: This change should not show up in the diff.

File: src/solver/conv_winoRxS.cpp [hunk 12]
```diff
@@ -709,11 +718,8 @@ ConvSolution ConvBinWinoRxS<Winodata, Winofilter>::GetSolution(
     kernel.l_wk.push_back(1);
     kernel.l_wk.push_back(1);
 
-    const auto force_cache_bypass = (name == "gfx940") || (name == "gfx941");
-
```
Comment (JehandadKhan) [inline src/solver/conv_winoRxS.cpp]: This change should not show up in the diff.

File: src/solver/conv_winoRxS.cpp [hunk 13]
```diff
@@ -709,11 +718,8 @@ ConvSolution ConvBinWinoRxS<Winodata, Winofilter>::GetSolution(
...
     KernelBuildParameters options{
         {"ROCM_METADATA_VERSION", 5},
-        {"FORCE_CACHE_BYPASS_ON_STORE", force_cache_bypass},
```
Comment (JehandadKhan) [inline src/solver/conv_winoRxS.cpp]: This change should not show up in the diff.

File: test/driver.hpp [hunk 14]
```diff
@@ -56,6 +56,23 @@ constexpr std::is_same<T, U> is_same(const T&)
     return {};
 }
 
+struct tensor_elem_gen_integer
```
Comment (JehandadKhan) [inline test/driver.hpp]: Please check if this change should come from your PR

**Review by JehandadKhan — CHANGES_REQUESTED** [review]

File: src/kernels/gpu_reference_kernel/fp8_naive_conv.cpp (hunk 6 above)
Comment (xinlipn) [reply src/kernels/gpu_reference_kernel/fp8_naive_conv.cpp]: Done

File: src/kernels/hip_f8_impl.h (hunk 7 above)
Comment (xinlipn) [reply src/kernels/hip_f8_impl.h]: Done

File: src/kernels/gpu_reference_kernel/fp8_kern_types.h (hunk 5 above)
Comment (xinlipn) [reply src/kernels/gpu_reference_kernel/fp8_kern_types.h:26]: Done

File: src/kernels/gpu_reference_kernel/fp8_kern_types.h (hunk 4 above)
Comment (xinlipn) [reply src/kernels/gpu_reference_kernel/fp8_kern_types.h]: Done

File: src/kernels/hip_float8.h (hunk 8 above)
Comment (xinlipn) [reply src/kernels/hip_float8.h]: All dead code removed

File: src/solver/conv_direct_naive_conv.cpp (hunk 9 above)
Comment (xinlipn) [reply src/solver/conv_direct_naive_conv.cpp]: This function is indeed a duplicate of 
inline std::string GetDataType(miopenDataType_t type)
in
miopen/datatype.hpp

File: test/driver.hpp (hunk 14 above)
Comment (xinlipn) [reply test/driver.hpp]: This struct is defined here on develop branch
test/tensor_holder.hpp

**Comment by xinlipn** [comment]
Will be replaced by this PR
https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2251

**Pull request closed without merging**